File: src/main/java/jugglinglab/notation/NotationControl.java
//
// NotationControl.java
//
// This is the GUI that allows the user to enter a pattern in a given notation.
// It is used by ApplicationPanel to assemble the interface.
//
// Copyright 2002-2025 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.notation;

import java.util.ResourceBundle;
import javax.swing.JPanel;
import jugglinglab.util.JuggleExceptionInternal;
import jugglinglab.util.JuggleExceptionUser;
import jugglinglab.util.ParameterList;

public abstract class NotationControl extends JPanel {
  static final ResourceBundle guistrings = jugglinglab.JugglingLab.guistrings;

  // Return a new (uninitialized) pattern in the target notation
  public abstract Pattern newPattern();

  // Return parameters defining Pattern, AnimationPrefs, and View.
  //
  // The control doesn't need to do any error-checking on whether parameters
  // are consistent, etc. -- those will be caught later when the parameters
  // are used to create a Pattern, AnimationPrefs, and so on.
  public abstract ParameterList getParameterList()
      throws JuggleExceptionUser, JuggleExceptionInternal;

  // Reset control to defaults
  public abstract void resetControl();
}

File: src/main/kotlin/jugglinglab/notation/NotationControl.kt
//
// NotationControl.kt
//
// This is the GUI that allows the user to enter a pattern in a given notation.
// It is used by ApplicationPanel to assemble the interface.
//
// Copyright 2002-2025 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.notation

import jugglinglab.util.JuggleExceptionInternal
import jugglinglab.util.JuggleExceptionUser
import jugglinglab.util.ParameterList
import javax.swing.JPanel

abstract class NotationControl : JPanel() {
    // Return a new (uninitialized) pattern in the target notation
    abstract fun newPattern(): Pattern

    @get:Throws(JuggleExceptionUser::class, JuggleExceptionInternal::class)
    abstract val parameterList: ParameterList

    // Reset control to defaults
    abstract fun resetControl()
}
